File: billingsoftware/src/main/java/in/vedchangani/billingsoftware/service/ItemService.java
package in.vedchangani.billingsoftware.service;

import in.vedchangani.billingsoftware.io.ItemRequest;
import in.vedchangani.billingsoftware.io.ItemResponse;
import in.vedchangani.billingsoftware.io.ItemUpdateRequest;
import in.vedchangani.billingsoftware.io.StockAdjustmentRequest;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface ItemService {

    ItemResponse add(ItemRequest request, MultipartFile file) throws IOException;

    List<ItemResponse> fetchItems();

    void deleteItem(String itemId);

    // General metadata edit. Never touches stockQuantity/reservedQuantity - see ItemUpdateRequest.
    ItemResponse update(String itemId, ItemUpdateRequest request);

    // Dedicated stock mutation path, backed by the atomic ItemRepository.adjustStockQuantity query.
    ItemResponse adjustStock(String itemId, StockAdjustmentRequest request);
}

File: client/src/Service/ItemService.js
import apiClient from "../util/apiClient.js";

export const addItem = async (item) => {
    return await apiClient.post(`/admin/items`, item);
}

export const deleteItem = async (itemId) => {
    return await apiClient.delete(`/admin/items/${itemId}`);
}

export const fetchItems = async () => {
    return await apiClient.get('/items');
}

// General metadata edit - never sends stockQuantity/reservedQuantity; stock only changes via
// adjustStock below.
export const updateItem = async (itemId, payload) => {
    return await apiClient.put(`/admin/items/${itemId}`, payload);
}

export const adjustStock = async (itemId, delta) => {
    return await apiClient.patch(`/admin/items/${itemId}/stock`, {delta});
}
